src/functions/template-tags/general.php

```diff
@@ -704,3 +704,100 @@ function tribe_is_frontend() {
 		return (bool) apply_filters( 'tribe_doing_frontend', false );
 	}
 }
+
+if ( ! function_exists( 'tribe_post_checksum' ) ) {
+	/**
+	 * Builds a post checksum based on a list of fields that are
+	 * set to change when relevant post details change.
+	 *
+	 * This function will only calculate the checksum: it will not perform any
+	 * kind of check or monitoring on the post itself.
+	 *
+	 * @since TBD
+	 *
+	 * @param int|WP_Post $post_id   The post ID or object
+	 * @param array       $use_frags An array detailing the post fields, columns of
+	 *                               the `posts` table, that should be used to build
+	 *                               the checksum; defaults to the post `ID` and `post_modified`
+	 *                               columns.
+	 *
+	 * @return null|string The checksum string or `null` if the post is not valid or no
+	 *                     fragments are passed.
+	 */
+	function tribe_post_checksum( $post_id, $use_frags = array( 'ID', 'post_modified' ) ) {
+		if ( empty( $post_id ) || empty( $use_frags ) ) {
+			return null;
+		}
+
+		$post = get_post( $post_id );
+
+		if ( ! $post instanceof WP_Post ) {
+			return null;
+		}
+
+		$frags = array();
+		foreach ( $use_frags as $field ) {
+			$frags[] = $post->{$field};
+		}
+
+		return md5( implode( '|', $frags ) );
+	}
+}
+
+if ( ! function_exists( 'tribe_posts_checksum' ) ) {
+	/**
+	 * Builds the checksum for a group of posts based on a list of fields that are
+	 * set to change when relevant post details change.
+	 *
+	 * This function will only calculate the checksum: it will not perform any
+	 * kind of check or monitoring on the posts themselves.
+	 *
+	 * @since TBD
+	 *
+	 * @param       array $post_ids  An array of post IDs or objects to calculate
+	 *                               the checksum for.
+	 * @param array       $use_frags An array detailing the post fields, columns of
+	 *                               the `posts` table, that should be used to build
+	 *                               the checksum; defaults to the post `ID` and `post_modified`
+	 *                               columns.
+	 *
+	 * @return null|string The checksum string or `null` if the posts are not valid or no
+	 *                     fragments are passed.
+	 */
+	function tribe_posts_checksum( $post_ids, $use_frags = array( 'ID', 'post_modified' ) ) {
+		$post_ids = array_filter( (array) $post_ids );
+
+		if ( empty( $post_ids ) ) {
+			return null;
+		}
+
+		/** @var wpdb $wpdb */
+		global $wpdb;
+
+		$prepared = array();
+		foreach ( $post_ids as $post_id ) {
+			if ( $post_id instanceof WP_Post ) {
```

Comment:
Why query at all if we already have access to post_modified?

Reply:
Right. I've updated the code to handle the case where the objects are all instances of `WP_Post`